source/lambda_functiontim.py

```diff
@@ -38,32 +44,46 @@ def setMetadata(formatted_tim, tim_dict):
+		tim_dict: Dict containing individual TIM from the Wyoming CV Pilot data file truncated to
 		include just metadata.
 
 	Returns:
-		formatted_tim with additional keys 
+		formatted_tim with additional keys
 	'''
-	formatted_tim['metadata_generatedAt'] = tim_dict['recordGeneratedAt'].replace('Z[UTC]','')
-	formatted_tim['metadata_recordGeneratedBy'] = tim_dict['recordGeneratedBy']
-	formatted_tim['metadata_logFileName'] = tim_dict['logFileName']
-	formatted_tim['metadata_securityResultCode'] = tim_dict['securityResultCode']
-	formatted_tim['metadata_sanitized'] = str(tim_dict['sanitized'])
-	formatted_tim['metadata_payloadType'] = tim_dict['payloadType']
-	formatted_tim['metadata_recordType'] = tim_dict['recordType']
-	formatted_tim['metadata_serialId_streamId'] = tim_dict['serialId']['streamId']
-	formatted_tim['metadata_serialId_bundleSize'] = tim_dict['serialId']['bundleSize']
-	formatted_tim['metadata_serialId_bundleId'] = tim_dict['serialId']['bundleId']
-	formatted_tim['metadata_serialId_recordId'] = tim_dict['serialId']['recordId']
-	formatted_tim['metadata_serialId_serialNumber'] = tim_dict['serialId']['serialNumber']
-	formatted_tim['metadata_receivedAt'] = tim_dict['odeReceivedAt'].replace('Z[UTC]','')
-	formatted_tim['metadata_rmd_elevation'] = tim_dict['receivedMessageDetails']['locationData']['elevation']
-	formatted_tim['metadata_rmd_heading'] = tim_dict['receivedMessageDetails']['locationData']['heading']
-	formatted_tim['metadata_rmd_latitude'] = tim_dict['receivedMessageDetails']['locationData']['latitude']
-	formatted_tim['metadata_rmd_longitude'] = tim_dict['receivedMessageDetails']['locationData']['longitude']
-	formatted_tim['metadata_rmd_speed'] = tim_dict['receivedMessageDetails']['locationData']['speed']
-	formatted_tim['metadata_rmd_rxSource'] = tim_dict['receivedMessageDetails']['rxSource']
-	formatted_tim['metadata_schemaVersion'] = tim_dict['schemaVersion']
+	schemaVersion = tim_dict.get('schemaVersion')
+	formatted_tim['metadata_schemaVersion'] = schemaVersion
+	formatted_tim['metadata_generatedAt'] = tim_dict.get('recordGeneratedAt', '').replace('Z[UTC]','')
+	formatted_tim['metadata_recordGeneratedBy'] = tim_dict.get('recordGeneratedBy')
+	formatted_tim['metadata_sanitized'] = str(tim_dict.get('sanitized', ''))
+	formatted_tim['metadata_payloadType'] = tim_dict.get('payloadType')
+	formatted_tim['metadata_serialId_streamId'] = tim_dict.get('serialId', {}).get('streamId')
+	formatted_tim['metadata_serialId_bundleSize'] = tim_dict.get('serialId', {}).get('bundleSize')
+	formatted_tim['metadata_serialId_bundleId'] = tim_dict.get('serialId', {}).get('bundleId')
+	formatted_tim['metadata_serialId_recordId'] = tim_dict.get('serialId', {}).get('recordId')
+	formatted_tim['metadata_serialId_serialNumber'] = tim_dict.get('serialId', {}).get('serialNumber')
+	formatted_tim['metadata_receivedAt'] = tim_dict.get('odeReceivedAt', '').replace('Z[UTC]','')
```

Comment:
Z[UTC] will no longer be in version 6. Will the absence break this code? Please make sure the code works with and without Z[UTC]

Reply:
The code will work without Z[UTC].